src/main/java/core/basesyntax/strategy/actions/PurchaseActionHandler.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.strategy.actions;
+
+import java.util.Map;
+import java.util.Optional;
+
+public class PurchaseActionHandler implements ActionHandler {
+    @Override
+    public void apply(Map<String, Integer> map, String data) {
+        String key = data.split(SPLITTER, 2)[0];
+        Integer purchasedAmount = Integer.parseInt(data.split(SPLITTER, 2)[1]);
```

Comment:
data.split(SPLITER,2) - put in local variable and use further where needed
make constants fo magic number 0,1 and 2
NumberFormatException - handle for parseInt